docs/architecture/modernize-desktop/windows-migration.md

```diff
@@ -0,0 +1,527 @@
+probably the right approach.
+
+### The road ahead XAML Islands: WinUI 3.0
+
+Since Windows 8, the Windows UI platform, including the XAML
+UI framework, visual composition layer, and input processing has been shipped as
+an integral part of Windows. This means that to benefit from the latest
+improvements on UI technologies you must upgrade to the latest version of the
+UI, slowing down the pace of innovation when you develop your apps. To decouple
+these two evolution cycles and foster innovation, Microsoft is actively working
+on the WinUI project.
+
+Starting with WinUI 2 in 2018, Microsoft start shipping some new XAML UI
+controls and features as separate NuGet packages that build on top of the UWP
+SDK.
+
+![WinUI 2.0](media\windows-migration\winui2.png)
+
+WinUI 3 is under active development and will greatly expand the scope of WinUI
+to include the full UI platform, which will now be fully decoupled from the UWP
+SDK:
+
+![WinUI 3.0](media\windows-migration\image36.png)
+
+XAML framework will now be developed on GitHub and shipped out
+of band as [NuGet](https://docs.microsoft.com/nuget/what-is-nuget) packages.
+
+The existing UWP XAML APIs that ship as part of the OS will no longer receive
+new feature updates. They will still receive security updates and critical fixes
+according to the Windows 10 support lifecycle.
+
+The Universal Windows Platform contains more than just the XAML framework (e.g.
+application and security model, media pipeline, Xbox and Windows 10 shell
+integrations, broad device support) and will continue to evolve. All new XAML
+features will just be developed and ship as part of WinUI instead.
+
+#### WinUI 3 in desktop app and WinUI XAML Islands
+
+As you can see, WinUI 3 is the evolution of UWP XAML and it works naturally
+within the UWP app model and all its requirements (MSIX packaged ID, sandbox,
+CoreWindow, etc). To use just WinUI 3 in a Win32 app model, the WinUI content
+should be hosted by another UI Framework (Windows Forms, WPF, etc.) using **WinUI
+XAML Islands**. This is the right path if you want to evolve your app and mix
+technologies. However, if you want to replace your entire old UI for WinUI, your
+app shouldn't load UI Frameworks for just hosting WinUI.
+
+WinUI 3 will address this critical feedback adding **WinUI in desktop apps**.
+This will allow that Win32 apps can use WinUI 3 as standalone UI Framework; no
+need to load Windows Forms or WPF.
+
+Within this aggregation, WinUI 3 will let developers easily mix and match the
+right combination of:
+
+* App model: UWP, Win32
+* Platform: .NET Core 3 or Native
+* Language: .NET (C\#, Visual Basic), standard C++
+* Packaging: MSIX, AppX for the Microsoft Store, unpackaged
+* Interop: use WinUI 3 to extend existing WPF, WinForms and MFC apps using WinUI
+  XAML Islands.
+
+If you want to know more details, Microsoft is sharing this roadmap in
+<https://github.com/microsoft/microsoft-ui-xaml>.
```

Comment:
should it link to https://github.com/microsoft/microsoft-ui-xaml/blob/master/docs/roadmap.md?